tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -57,38 +57,41 @@ class TlaCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter
     if (state.isEmpty) {
       pretty.write(ValEx(TlaBool(true)))
     } else {
-      state.toList.sortBy(_._1).foreach{
-        case (name,value) =>
-          writer.print(s"/\\ $name = " )
-          pretty.write(value)
-          writer.println
+      state.toList.sortBy(_._1).foreach { case (name, value) =>
+        writer.print(s"/\\ $name = ")
+        pretty.write(value)
+        writer.println
       }
     }
   }
 
-  override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState] ) : Unit = {
+  override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState]): Unit = {
     val pretty = new PrettyWriter(writer)
 
     writer.println("%s MODULE counterexample %s\n".format("-" * 25, "-" * 25))
     writer.println("EXTENDS %s\n".format(rootModule.name))
 
     states.zipWithIndex.foreach {
+      case (state, 0) =>
+        writer.println("(* Constant initialization state *)")
```

Comment:
It is strange that we mix `PrettyWriter` and `PrintWriter` in this code. `PrettyWriter` may have its own print buffer and it does know when to flush the buffer. I would prefer to extend `PrettyWriter.toDoc(mod: TlaModule)` with an optional parameter that receives the modules to be extended.

Of course, that will require some fiddling with kyama `PrettyPrinter`, but the result is rewarding. The closest code is probably in `CASE`, though you will need "," instead of "[]" as a separator:

```scala
        val doc = group(text("CASE") <> nest(space <> folddoc(pairsWithOther.toList, _ <> line <> "[]" <> space <> _)))
```

Reply:
I have checked `PrettyWriter` and you must be safe w.r.t. buffers as long as you call `write` methods. However, moving the `EXTENDS` code to `PrettyWriter.toDoc` is still valuable, as you can reuse it.

Reply:
I noticed that it was using both, but I assumed you must have had some reason to do it this way at the time, so I just extended the extra case in the same way.
As for your `.toDoc` method suggestion, I'm not exactly sure what you're trying to do, could you elaborate more?

Reply:
I suggested to move the `EXTENDS...` code into `PrettyWriter`. That means that you have to extend the method `toDoc` for `TlaModule`.